packages/salesforcedx-vscode-lwc/src/testSupport/testRunner/testRunner.ts

```diff
@@ -94,9 +103,7 @@ export class TestRunner {
     } else {
       runTestsByPathArgs = [];
     }
-    const testNamePatternArgs = testName
-      ? ['--testNamePattern', `${escapeStrForRegex(testName)}`]
-      : [];
+    const testNamePatternArgs = getTestNamePatternArgs(testExecutionInfo);
```

Comment:
dumb q: why not just pass testExecutionInfo.testName here?

Reply:
Not dumb - I was actually kind of unclear why this specific check was happening on the name, and it was a little overly verbose to pass this in as a whole:
`'testName' in testExecutionInfo ? testExecutionInfo.testName : undefined;`

Reply:
gotcha..I'm a sicko so I had to go take a look.  Basically the way types are being used here is pretty weird.  It seems clearer to add at typeguard here ala 

```
// Probably could go in types file. 
function isTestCaseInfo(
  testExecutionInfo: TestExecutionInfo
): testExecutionInfo is TestCaseInfo {
  return testExecutionInfo.kind === TestInfoKind.TEST_CASE;
}

export function getTestNamePatternArgs(testName: string) {
  return ['--testNamePattern', `${escapeStrForRegex(testName)}`];
}

...  
const testNamePatternArgs =
       // Is the type a TestCaseInfo and is testName not emptyString
      (isTestCaseInfo(testExecutionInfo) && testExecutionInfo.testName)
        ? getTestNamePatternArgs(testExecutionInfo.testName)
        : [];
```

Reply:
That check was happening b/c the type `export type TestExecutionInfo = TestCaseInfo | TestFileInfo | TestDirectoryInfo;` is kind of silly.  testName only exists on the `TestCaseInfo` type FYI